File: DoItTest.Domain/Tests/TestInfo.cs
namespace DoItTest.Domain.Tests
{
    public class TestInfo
    {
        public Guid TestId { get; }
        public String Title { get; }
        public Int32 TimeToCompleteInSeconds { get; }
        public Int32 TestItemCount { get; }
        
        public TestInfo(Guid testId, String title, Int32 timeToCompleteInSeconds, Int32 testItemCount)
        {
            TestId = testId;
            Title = title;
            TimeToCompleteInSeconds = timeToCompleteInSeconds;
            TestItemCount = testItemCount;
        }
    }
}

File: DoItTest/ClientApp/src/domain/tests/testInfo.ts
export class TestInfo {
    public constructor(
        public readonly testId: string,
        public readonly title: string,
        public readonly timeToCompleteInSeconds: number,
        public readonly testItemCount: number
    ) { }
}

export function mapToTestInfo(value: any): TestInfo {
    return new TestInfo(value.testId, value.title, value.timeToCompleteInSeconds, value.testItemCount)
}